R/derive_param_bor.R

```diff
@@ -0,0 +1,541 @@
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%|
+#' @description
+#'     Derives Best Overall Response
+#++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @details
+#'    Calculates the best overall response (BOR) parameter, as detailed below.
+#'
+#'    Records after PD can be removed using the source_pd and source_datasets
+#'    arguments.
+#'
+#'   Note:
+#'
+#'   1. All records where `ADT` >= `reference_date` + `ref_start_window` are considered
+#'      for Best Overall Response.
+#'
+#'   2. All records where `ADT` < `reference_date` + `ref_start_window`  and the subject
+#'      has a record in step 1, are considered for Best Overall Response.
+#'
+#'   3. Records, where `ADT` < `reference_date` + `ref_start_window` and the subject does
+#'      not have a record in step 1 but AVALC %in% c("CR", "PR", "PD"), are considered for
+#'      Best Overall Response
+#'
+#'   4. Records where `ADT` < `reference_date` + `ref_start_window` and the subject does
+#'      not have a record in step 1 and AVALC %in% c('SD', 'NON-CR/NON-PD') then
+#'      Best Overall Response is set to NE.
```

Comment:
i find these steps 1-4 confusingly written. might be better something like:

```
#'   1. All `CR`, `PR` and `PD` response records are considered for Best Overall Response.
#'
#'   2. All `SD` or `NON-CR/NON-PD` records where `ADT` >= `reference_date` +
#'      `ref_start_window` are also considered for Best Overall Response.
#'
#'   3. Patients with **ONLY** `SD` or `NON-CR/NON-PD` records where `ADT` <
#'      `reference_date` + `ref_start_window` would give Best Overall Response of `NE`.
```

Reply:
Yes, my misunderstanding as we discussed, updated now.